packages/nimble-components/src/table/specs/table-column-interaction-events.md

```diff
@@ -0,0 +1,60 @@
+# Table Column Interaction Events HLD
+
+## Problem Statement
+
+Some clients need to know when a user interactively changes the configuration of their table columns. This includes:
+
+-   Sorting a column
+-   Grouping by a column
+-   Resizing a column
+
```

Comment:
Other possibilities to consider here are:
- reordering columns (included in wireframes via drag-drop)
- changing column visibility (was a feature of ext grid)

I'd vote to list them here even if we don't have any immediate plans to implement them.